src/test/scala/io/shiftleft/fuzzyc2cpg/astnew/AstToCpgTests.scala

```diff
@@ -15,30 +17,44 @@ import org.scalatest.{Matchers, WordSpec}
 
 class AstToCpgTests extends WordSpec with Matchers {
 
-  private class GraphAdapter(graph: ScalaGraph) extends CpgAdapter[Vertex, Vertex] {
+  private class GraphAdapter(graph: ScalaGraph) extends CpgAdapter[Vertex, Vertex, Edge, Edge] {
     override def createNodeBuilder(kind: NodeKind): Vertex = {
       graph.addVertex(kind.toString)
     }
 
+    override def createNode(vertex: Vertex, origAstNode: AstNode): Vertex = {
+      vertex
+    }
+
     override def createNode(vertex: Vertex): Vertex = {
       vertex
     }
 
-    override def addProperty(vertex: Vertex, property: NodeProperty, value: String): Unit = {
+    override def addNodeProperty(vertex: Vertex, property: NodeProperty, value: String): Unit = {
       vertex.property(property.toString, value)
     }
 
-    override def addProperty(vertex: Vertex, property: NodeProperty, value: Int): Unit = {
+    override def addNodeProperty(vertex: Vertex, property: NodeProperty, value: Int): Unit = {
       vertex.property(property.toString, value)
     }
 
-    override def addProperty(vertex: Vertex, property: NodeProperty, value: Boolean): Unit = {
+    override def addNodeProperty(vertex: Vertex, property: NodeProperty, value: Boolean): Unit = {
       vertex.property(property.toString, value)
     }
 
-    override def addEdge(edgeKind: EdgeKind, dstNode: Vertex, srcNode: Vertex): Unit = {
-      srcNode.addEdge(edgeKind.toString, dstNode)
+    override def createEdgeBuilder(dst: Vertex, src: Vertex, edgeKind: EdgeKind): Edge = {
+      src.addEdge(edgeKind.toString, dst)
     }
+
+    override def createEdge(edge: Edge): Edge = {
+      edge
+    }
+
+    // Not yet implemented because not yet used.
```

Comment:
If they are not implemented, can we remove them?

Reply:
I removed the two unused versions of `addEdgeProperty`.
The test adapter implementations do not need to implement all methods because not all of them are relevant for the test.